sdk/storage/Azure.Storage.Blobs/src/BlobContainerClient.cs

```diff
@@ -1006,18 +1005,14 @@ private void SetNameFieldsIfNull()
                     // Return an exploding Response on 304
                     if (response.IsUnavailable())
                     {
-                        return response.GetRawResponse().AsNoBodyResponse<BlobContainerItem>();
+                        return response.GetRawResponse().AsNoBodyResponse<BlobContainerProperties>();
                     }
 
-                    // Turn the flattened properties into a BlobContainerItem
-                    var uri = new BlobUriBuilder(Uri);
+                    // Turn the flattened properties into a BlobContainerProperties
                     return Response.FromValue(
-                        new BlobContainerItem(false)
-                        {
-                            Name = uri.BlobContainerName,
```

Comment:
Yeah, this makes sense since we already have `containerClient.Name` with the same info.